crates/driver/src/infra/blockchain/token.rs

```diff
@@ -207,6 +167,150 @@ impl Erc20 {
     }
 }
 
+/// Uses a custom helper contract to simulate balances while taking
```

Comment:
I think this comment should move into the implementation (since the trait doesn't define how the balance is simulated). Also I think our future selfs would be happy to have a bit more context on how the `ZkSyncTradableBalanceSimulator` works and why this differentiation is needed.